JCA-63: Rond documentation review (#60)
rond-authz/rond-authz.github.io: _docs/configuration.md

#### Specify API permissions

In order to define the Rego policies to be evaluated for the API to be authorized, you must define the Rönd custom attribute in your OAS schema.  
Depending on the version of Rönd you are using, you have define `x-rond` or `x-permission` attirbutes.

**For version 1.5 or newer**, Rönd lets you define the `x-rond` attribute in your OAS schema.
This attribute is an object that contains the following properties:
```json
"x-rond": {
    "requestFlow": {
        "policyName": "generate_filter",
        "generateQuery": true,
        "queryOptions": {
            "headerName": "x-query-header"
        }
    },
    "responseFlow": {
        "policyName": "responsepolicy"
    },
    "options": {

>         "enableResourcePermissionsMapOptimization": false

Reviewer: Qui mancano le descrizioni

Author: improved descriptions for x-rond fields

    }
}
```

**For older versions** the `x-permission` attribute should be used.  
This attribute is shaped as an object with the following properties:
```json
"x-permission": {
    "allow": "name_of_the_policy_to_execute_upon_API_invocation", // Required.
    "resourceFilter": {
        "rowFilter": {
            "enabled": true,
            "headerKey": "acl_rows" // Identifier of the header sent to the requested service. The interpolated query will be injected Here. Defaults to `acl_rows`.
        }
    },
    "responseFilter": {

>         "policy": "name_of_the_policy_to_execute_upon_API_invocation"

Reviewer: In realtà è il nome della policy che viene eseguita in risposta. Non riporterei la stessa descrizione della policy allow

Reviewer 2: ```suggestion
        "policy": "name_of_the_policy_to_execute_during_API_response"
```

**For version 1.5 or newer**, Rönd lets you define the `x-rond` attribute in your OAS schema.
This attribute is an object that contains the following properties:
```json
"x-rond": {
    "requestFlow": {
        "policyName": "policy_to_be_executed_BEFORE_API_invocation",
        "generateQuery": true,
        "queryOptions": {
            "headerName": "x-query-header"
        }
    },
    "responseFlow": {
        "policyName": "policy_to_be_executed_AFTER_API_invocation"
    },
    "options": {
        "enableResourcePermissionsMapOptimization": false
    }
}
```

Here, `requestFlow` and `responseFlow` fields, define which policies have to be executed repectively before and after the API invocation.

{%
  include alert.html
  type="info"
  content="The `generateQuery` flag defines wether or not a query has to be generated and then forwarded to the application service through the header specified in the `headerName` field."
%}

**For older versions** the `x-permission` attribute should be used.  
This attribute is shaped as an object with the following properties:
```json
"x-permission": {

>     "allow": "name_of_the_policy_to_execute_upon_API_invocation", // Required.

Author: ```suggestion
    "allow": "name_of_the_policy_to_execute_BEFORE_API_invocation", // Required.
```

    "resourceFilter": {
        "rowFilter": {
            "enabled": true,
            "headerKey": "acl_rows" // Identifier of the header sent to the requested service. The interpolated query will be injected Here. Defaults to `acl_rows`.
        }
    },
    "responseFilter": {

>         "policy": "name_of_the_policy_to_execute_upon_API_invocation"

Author: ```suggestion
        "policy": "name_of_the_policy_to_execute_AFTER_API_invocation"
```

    },
    "options": {
        "enableResourcePermissionsMapOptimization": false
    },
}
```

{%
  include alert.html
  type="info"
  content="The `enableResourcePermissionsMapOptimization` flag enables the generation of an optimized map of user permissions. This is useful when performing RBAC logics.  
  More information available in the [Policy Integration page](policy-integration)"
%}

{%
  include alert.html
  type="warning"

>   content="For Rönd versions **older than 1.5**, when defining the `resourceFilter` attribute, Rönd changes the behavior of the `allow` policy.  

Reviewer 2: Why older than 1.5? I mean, the field had this name before but the feature is still the same; I think this alert is a bit misleading

Author: @fredmaggiowski shall i remove just the final part of the sentence?
"Rönd changes the behavior of the `allow` policy."

Author: ```suggestion
  content="For Rönd versions **older than 1.5**, you should define the `resourceFilter` attribute.  
```

    "paths": {
        "/hello": {
            "get": {

>                 "x-rond": {

Reviewer 2: x-permission and x-rond have different schemas, we should provide both in the docs, giving more focus to the new one (x-rond) and telling from which version it is available. x-permission should be still documented for older versions.

[Here is an example of the newer configuration format](https://github.com/rond-authz/rond/blob/main/mocks/rondOasConfig.json#L5)

Author: both x-permission and x-rond are now documented
